packages/react-storage/src/components/StorageBrowser/views/LocationActionView/DownloadView/useDownloadView.ts

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { isFunction } from '@aws-amplify/ui';
+
+import type { FileDataItem } from '../../../actions';
+import type { Task } from '../../../tasks';
+import { useLocationItems } from '../../../locationItems';
+import { useStore } from '../../../store';
+import { useAction } from '../../../useAction';
+
+import type { DownloadViewState, UseDownloadViewOptions } from './types';
+
+// assign to constant to ensure referential equality
+const EMPTY_ITEMS: FileDataItem[] = [];
+
+export const useDownloadView = (
+  options?: UseDownloadViewOptions
+): DownloadViewState => {
+  const { onExit: _onExit } = options ?? {};
+
+  const [{ location }, storeDispatch] = useStore();
+  const [locationItems, locationItemsDispatch] = useLocationItems();
+  const { current } = location;
+  const { fileDataItems: items = EMPTY_ITEMS } = locationItems;
+
+  const [processState, handleProcess] = useAction('download', {
+    items,
+  });
+
+  const { isProcessing, isProcessingComplete, statusCounts, tasks } =
+    processState;
+
+  const onActionStart = () => {
+    if (!current) return;
+    handleProcess();
+  };
+
+  const onActionCancel = () => {
+    tasks.forEach((task) => {
+      // @TODO Fixme, calling cancel on task doesn't currently work
```

Comment:
Very nit, this is more common in the codebase
```suggestion
      // TODO: calling cancel on task doesn't currently work
```

Reply:
Not sure this TODO is actually relevant. "cancel" should work as long as the task has a status of "QUEUED"

Reply:
+1. This is copied over code from delete. So bascially taskRemove handle removing the queue. I think this is more once you start the download/delete/copy if you can cancel it. Which will not work. So technically i think we want to disable the button in those use cases